crates/lsp/src/lib.rs

```diff
@@ -282,30 +310,39 @@ impl<L: LSPLang> Backend<L> {
       map: DashMap::new(),
     }
   }
-  async fn publish_diagnostics(&self, uri: Url, versioned: &VersionedAst<StrDoc<L>>) -> Option<()> {
+  async fn get_diagnostics(
+    &self,
+    uri: &Url,
+    versioned: &VersionedAst<StrDoc<L>>,
+  ) -> Option<Vec<Diagnostic>> {
     let mut diagnostics = vec![];
     let path = uri.to_file_path().ok()?;
 
-    let rules = match &self.rules {
-      Ok(rules) => rules.for_path(&path),
-      Err(_) => {
-        return Some(());
-      }
-    };
+    let rules = self.rules.as_ref().ok()?.for_path(&path);
```

Comment:
This is not the same as the previous code. 
if rules is Err then the get_diagnosticsw returns None